v2/dtos/requests/event.go

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package requests
+
+import (
+	"encoding/json"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2"
+	dtoBase "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common/base"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+// AddEventRequest defines the Request Content for POST event DTO. This object and its properties correspond to the
+// AddEventRequest object in the APIv2 specification.
+type AddEventRequest struct {
+	dtoBase.Request `json:",inline"`
+	Device          string        `json:"device" codec:"device,omitempty" validate:"required"`
+	Origin          int64         `json:"origin,omitempty" codec:"origin,omitempty" validate:"gte=1"`
+	Readings        []BaseReading `json:"readings,omitempty" codec:"readings,omitempty"`
+}
+
+// UpdateEventPushedByChecksumRequest defines the Request Content for PUT event as pushed DTO. This object and its properties correspond to the
+// UpdateEventPushedByChecksumRequest object in the APIv2 specification.
+type UpdateEventPushedByChecksumRequest struct {
+	dtoBase.Request `json:",inline"`
+	Checksum        string `json:"checksum"`
+}
+
+// Validate satisfies the Validator interface
+func (a AddEventRequest) Validate() error {
+	v2.NewValidator()
+	err := v2.Validate(a)
+	return err
+}
+
+func (a *AddEventRequest) UnmarshalJSON(b []byte) error {
+	var addEvent struct {
+		dtoBase.Request
+		Device   string
+		Origin   int64
+		Readings []BaseReading
+	}
+	if err := json.Unmarshal(b, &addEvent); err != nil {
+		return v2.NewErrContractInvalid("Request body type is not a JSON array.")
+	}
+
+	a.RequestID = addEvent.RequestID
+	a.CorrelationID = addEvent.CorrelationID
+	a.Device = addEvent.Device
+	a.Origin = addEvent.Origin
+	a.Readings = addEvent.Readings
+
+	// validate AddEventRequest DTO
+	if err := a.Validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ToEventModels(addRequests []AddEventRequest) (events []models.Event) {
```

Comment:
We need to set up a naming convention.  Event can't be updated, but we will have more data models can be updated, such as Device.
Thus, I suggest to use `AddReqToEventModels`
We will have something like `AddReqtToDeviceModels `and `UpdateReqtToDeviceModels` in the future.